src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -58,22 +52,6 @@ CHIP_ERROR CASEServer::InitCASEHandshake(Messaging::ExchangeContext * ec)
 {
     ReturnErrorCodeIf(ec == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
-    // Mark any PASE sessions used for commissioning as stale.
-    // This is a workaround, as we currently don't have a way to identify
-    // secure sessions established via PASE protocol.
-    // TODO - Identify which PASE base secure channel was used
-    //        for commissioning and drop it once commissioning is complete.
-    mSessionManager->ExpireAllPairings(kUndefinedNodeId, kUndefinedFabricIndex);
-
-#if CONFIG_NETWORK_LAYER_BLE
-    // Close all BLE connections now since a CASE handshake has been initiated.
-    if (mBleLayer != nullptr)
-    {
-        ChipLogProgress(Discovery, "CASE handshake initiated, closing all BLE Connections");
```

Comment:
We still need to do this on the device side (but not in this code), right? See "Proposed solution" section of https://github.com/project-chip/connectedhomeip/issues/16766#issue-1184914983

Generally, it seems like we should do the BLE bits in a separate PR from the PASE session bits here....

Reply:
Are you requesting a change here? Or just that the change be separated into another PR? The BLE changes are handled in a separate commit.

Reply:
I would prefer the BLE bits in a separate PR, because right now they are regressing behavior (not actually doing the BLE bits on the server, where we do the right now) and I don't want to hold up the other parts on the BLE part...

Reply:
Can you clarify the regression? This code block is just moved out to the device commissioner so we don't close the PASE connection as soon as we start initiating CASE. The logic in the CASE server is incorrect and blocks the ability to recover from errors using the PASE connection if we fail to establish CASE.

Reply:
In the current setup (which has some problems, as you note), the idea is that on the _commissionee_ once CASE is established the BLE connection will be closed.  This code is not "supposed" to be running on the commissioner at all, because the commissioner wasn't even running a CASEServer when this code was added.  

So moving it to device commissioner totally changes who is closing what, right?

I can see that you need to change the BLE stuff in CASEServer if we're going to not close the BLE connection on either side until after commissioning is complete.  What we should be doing is something like what https://github.com/project-chip/connectedhomeip/issues/16766 describes.

1) The _commissionee_ closes it at the right point (when commissioning is complete, or the fail-safe expires, or the commissioning window closes and there is no existing PASE session).
2) The _commissioner_ closes it at the right point (releasing commissionee device seems ok).  And even then, closing _all_ commiss

I guess given that this CASEServer bit blocks you it might be OK to implement just item 2 for now and then file a followup for item 1, but we need to make sure it happens.

Reply:
OK, I see what you're saying.

The commissioning window manager already closes out all the BLE connections on a failed attempt and we do expire the pase pairings on commissioning complete, but the BLE connections don't get closed. If we close the BLE connections on commissioning complete, that seems like it would solve the issue. Thoughts?

Reply:
Yes, that would solve the issue related to CASE.

Looks like we already close BLE connections on session establishment error (as you said) or fail-safe expiration.

The one place that would be left is when the commissioning window timer expires before a PASE session is established.  We don't arm the "initial" fail-safe timer until PASE establishment, so it does not help here.  This should definitely be a separate follow-up for this PR.

Reply:
added in next commit, tracking issue for no PASE connection is here: https://github.com/project-chip/connectedhomeip/issues/17508